cmd/autoconfigbrancher/main.go

```diff
@@ -205,4 +209,48 @@ func main() {
 		matchTitle, o.githubLogin+":"+remoteBranch, "master"); err != nil {
 		logrus.WithError(err).Fatal("PR creation failed.")
 	}
+
+	if !o.selfApprove {
+		return
+	}
+
+	if err := selfApprove(gc, githubOrg, githubRepo); err != nil {
+		logrus.WithError(err).Fatal("Self-approve failed.")
+	}
+
+}
+
+func selfApprove(gc github.Client, org, repo string) error {
+	logrus.Infof("Self-aproving PR by adding the %q and %q labels", labels.Approved, labels.LGTM)
+
+	me, err := gc.BotName()
+	if err != nil {
+		return fmt.Errorf("bot name: %v", err)
+	}
+
+	issues, err := gc.FindIssues("is:open is:pr archived:false in:title author:"+me+" "+matchTitle, "updated", false)
+	if err != nil {
+		return fmt.Errorf("find issues: %v", err)
+	} else if len(issues) == 0 {
+		logrus.Info("No reusable issues found")
+		return nil
```

Comment:
Can we be super-paranoid and also bail out on >1? That would mean we have an accidental match which we probably do not want to auto-approve.

Reply:
Updated